web/src/main/java/org/fao/geonet/kernel/harvest/harvester/ogcwxs/Harvester.java

```diff
@@ -520,7 +521,17 @@ private WxSLayerRegistry addLayerMetadata (Element layer, Element capa) throws J
 			reg.name 	= layer.getChild ("name", wcs).getValue ();
 		} else if (params.ogctype.substring(0,3).equals("SOS")) {
 			Namespace gml = Namespace.getNamespace("http://www.opengis.net/gml");
-			reg.name 	= layer.getChild ("name", gml).getValue ();
+                        if(layer.getChild ("name", gml) != null) {
+                            reg.name 	= layer.getChild ("name", gml).getValue ();
+                        }
+                        else if(layer.getAttribute("id", gml) != null) {
+                            reg.name    = layer.getAttribute("id", gml).getValue();
+                        }
+                        else {
```

Comment:
Naturally, creating a random layer name does not allow accessing the layer in the layer parsing.